server/internal/item_internal/components.go

```diff
@@ -0,0 +1,84 @@
+package item_internal
+
+import (
+	"github.com/df-mc/dragonfly/server/item"
+	"github.com/df-mc/dragonfly/server/world"
+	"strings"
+)
+
+// Components returns all the components of the given custom item. If the item has no components, a nil map and false
+// are returned.
+func Components(it world.CustomItem) (map[string]interface{}, bool) {
+	category := it.Category()
+	identifier, _ := it.EncodeItem()
+	name := strings.Split(identifier, ":")[1]
+
+	builder := NewComponentBuilder(it.Name(), identifier, category)
+
+	if x, ok := it.(item.Armour); ok {
+		builder.AddComponent("minecraft:armor", map[string]interface{}{
+			"protection": int32(x.DefencePoints()),
+		})
+		builder.AddComponent("minecraft:knockback_resistance", map[string]interface{}{
```

Comment:
Still a bunch of `interface{}` here instead of `any`, would just do a replace all in dragonfly with interface{} => any.